### PR TITLE
RSARTE-176558: Fixed static analyzer findings on LibTcpServer

### PR DESCRIPTION
**Pull request by rajneshkumarmalik**
Fixed static analyzer findings on LibTcpServer

### CODE REVIEW

File: LibTCPServer.emx [hunk 1]
```diff
@@ -402,7 +402,7 @@ delete job;]]></body>
 		const RTPortDescriptor* portDescriptor = actorData->ports;
 		for (int i = actorData->numPorts; --i >= 0 && !foundPortDescriptor; ++portDescriptor ) {
 			RTProtocol* port = RTObjectOf( RTProtocol, portDescriptor->offset, this );
-			if (portName == port->getName()) {
+			if (!RTMemoryUtil::strcmp(portName.c_str(), port->getName())) {
```
Comment (mattias-mohlin) [inline LibTCPServer.emx]: @rajneshkumarmalik What is the reason for this change? There is an operator== defined on std::string for comparing with a const char* string. Was it not invoked here?

File: LibTCPServer.emx [hunk 2]
```diff
@@ -876,7 +876,8 @@ if (config.logToStdOut)
 		bool firstPort = true;
 		for (int i = actorData->numPorts; --i >= 0; ++portDescriptor ) {
 			RTProtocol* port = RTObjectOf( RTProtocol, portDescriptor->offset, this );
-			if (port->getName() == "external" && port->getInfo()->protocol->name == "External")
+			if (!RTMemoryUtil::strcmp(port->getName(), "external") &&
```
Comment (mattias-mohlin) [inline LibTCPServer.emx:879]: Same question as above

File: LibTCPServer.emx [hunk 3]
```diff
@@ -934,6 +935,9 @@ if (config.logToStdOut)
         </ownedParameter>
       </ownedOperation>
       <ownedOperation xmi:id="_j8WecPF5EeiGPObbQ5S9rA" name="unexpectedMessage" method="_29hpUPF5EeiGPObbQ5S9rA">
+        <eAnnotations xmi:id="_qUbSEE2oEe-BorRbVt8CFA" source="com.hcl.xtools.properties">
+          <details xmi:id="_qUb5IE2oEe-BorRbVt8CFA" key="cpp/overrideFunction/java.lang.Boolean" value="true"/>
```
Comment (mattias-mohlin) [inline LibTCPServer.emx:939]: ok

**Review by mattias-mohlin — CHANGES_REQUESTED** [review]
Please check the question above.

File: LibTCPServer.emx (hunk 1 above)
Comment (rajneshkumarmalik) [reply LibTCPServer.emx]: Clang compiler was showing warning for this scenario. After searching on internet I find out that It is not supported by Clang compiler. So, this fix was made to keep Clang compiler happy. Kindly let me know your feedback.

Comment (mattias-mohlin) [reply LibTCPServer.emx]: @rajneshkumarmalik Which warning did you get? Do you mean to say that the std::string implementation in Clangd doesn't have this operator== defined? What happens when you step into that function in the debugger?
IMO the new code is less readable than the original one (more C than C++), so we need to understand what is the reason here.

Comment (rajneshkumarmalik) [reply LibTCPServer.emx]: I got following warning.
/TCPServer.cpp:821:24: warning: result of comparison against a string literal is unspecified (use an explicit string comparison function instead) [-Wstring-compare]
                        if (port->getName() == "external" && port->getInfo()>protocol>name == "External")
                        
When I visit llvm github repo [https://github.com/llvm/llvm-project/blob/main/libcxx/include/string] I find out some comment to remove support in C++20. I think it is not present for all C++ version in clang now. I tried to debug but somehow control is not hitting this function. 
bool operator==(const charT* lhs, const basic_string<charT, traits, Allocator>& rhs) noexcept;  // removed in C++20

Comment (mattias-mohlin) [reply LibTCPServer.emx]: @rajneshkumarmalik Yes, for that code I can understand the warning because here we are incorrectly comparing two const char pointers instead of doing string comparison. But here we are talking about this code
```cpp
portName == port->getName()
```
where the left side is an std::string and the right side is a const char *.
What warning do you get for the line of code?

File: LibTCPServer.emx (hunk 2 above)
Comment (mattias-mohlin) [reply LibTCPServer.emx:879]: Here the warning is understandable since we compare two char pointers instead of strings. So this change looks correct.

File: LibTCPServer.emx (hunk 1 above)
Comment (rajneshkumarmalik) [reply LibTCPServer.emx]: I have rechecked it again. In this case warning is not observed. I might fix this code to maintain uniformity and assuming that in future we might get warning for this piece of code. Warning is observed in below case only. 
Kindly let me know that I can revert this code or keep it.

Comment (mattias-mohlin) [reply LibTCPServer.emx]: @rajneshkumarmalik If there is no warning we shouldn't change this code. The new code is less readable than the original one, so please revert it.

Comment (rajneshkumarmalik) [reply LibTCPServer.emx]: Done.

**Review by mattias-mohlin — APPROVED** [review]
ok
